source/common/upstream/upstream_impl.h

```diff
@@ -490,6 +494,37 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
   uint64_t pending_initialize_health_checks_{};
 };
 
+/**
+ * Manage priority state of a cluster.
```

Comment:
Do we comment what the PriorityState of a cluster is anywhere?  I don't see comments on the typedef and I'd encourage explaining in a bit more depth here or there.